### PR TITLE
[MISC] Move seqan3::views::single_pass_input to seqan3/utility/views/

### PR DESCRIPTION
**Pull request by simonsasse**
Resolves: https://github.com/seqan/product_backlog/issues/360 

There was no patch in the issue description so I did not apply any patch.

### CODE REVIEW

**Review by SGSSGene — APPROVED** [review]
:+1:

**Review by SGSSGene — CHANGES_REQUESTED** [review]
You need to move "test/snippet/range/views/single_pass_input.cpp" to test/snippet/utility...

**Comment by simonsasse** [comment]
Thanks for the review. I rebased, moved the snippet as you demanded. I also moved the imports starting with seqan3/test back to the position. I separated them from the other seqan3 imports, because I thought it would correct. 👍

**Comment by simonsasse** [comment]
I also need to move the seqan3/test/unit/range/views/single_pass_input_test.cpp to seqan3/test/unit/utility/views/single_pass_input_test.cpp.
I'll reassign then !

**Review by SGSSGene — APPROVED** [review]
:+1:

File: include/seqan3/utility/views/single_pass_input.hpp [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ * \brief Provides seqan3::single_pass_input_view
+ */
+
+#pragma once
+
+#include <seqan3/range/views/detail.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/iterator>
+#include <seqan3/std/ranges>
```
Comment (marehr) [inline include/seqan3/utility/views/single_pass_input.hpp:18]:
```suggestion
#include <seqan3/std/concepts>
#include <seqan3/std/iterator>
#include <seqan3/std/ranges>

#include <seqan3/range/views/detail.hpp>
```

File: test/snippet/range/views/take_until.cpp [hunk 2]
```diff
@@ -2,9 +2,9 @@
 #include <string>
 
 #include <seqan3/core/debug_stream.hpp>                 // for debug_stream
-#include <seqan3/range/views/single_pass_input.hpp>     // for views::single_pass_input
 #include <seqan3/range/views/take_until.hpp>            // for views::take_until*
 #include <seqan3/utility/char_operations/predicate.hpp> // for is_char
+#include <seqan3/utility/views/single_pass_input.hpp>     // for views::single_pass_input
```
Comment (marehr) [inline test/snippet/range/views/take_until.cpp:7]:
```suggestion
#include <seqan3/utility/views/single_pass_input.hpp>  // for views::single_pass_input
```

Comment (marehr) [inline test/snippet/range/views/take_until.cpp:7]:
```suggestion
#include <seqan3/utility/char_operations/predicate.hpp> // for is_char
#include <seqan3/utility/views/single_pass_input.hpp>   // for views::single_pass_input
```

**Review by marehr — APPROVED** [review]

**Comment by marehr** [comment]
rebase needed :)

**Pull request merged**